src/app/tests/TestDefaultICDClientStorage.cpp

```diff
@@ -200,6 +204,55 @@ void TestClientInfoCountMultipleFabric(nlTestSuite * apSuite, void * apContext)
     NL_TEST_ASSERT(apSuite, count == 0);
 }
 
+void TestProcessCheckInPayload(nlTestSuite * apSuite, void * apContext)
+{
+    CHIP_ERROR err       = CHIP_NO_ERROR;
+    FabricIndex fabricId = 1;
+    NodeId nodeId        = 6666;
+    TestPersistentStorageDelegate clientInfoStorage;
+    TestSessionKeystoreImpl keystore;
+
+    DefaultICDClientStorage manager;
+    err = manager.Init(&clientInfoStorage, &keystore);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    err = manager.UpdateFabricList(fabricId);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    // Populate clientInfo
+    ICDClientInfo clientInfo;
+    clientInfo.peer_node = ScopedNodeId(nodeId, fabricId);
+
+    err = manager.SetKey(clientInfo, ByteSpan(kKeyBuffer1));
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    err = manager.StoreEntry(clientInfo);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    uint32_t counter                  = 1;
+    System::PacketBufferHandle buffer = MessagePacketBuffer::New(chip::Protocols::SecureChannel::CheckinMessage::sMinPayloadSize);
+    MutableByteSpan output{ buffer->Start(), buffer->MaxDataLength() };
+    err = chip::Protocols::SecureChannel::CheckinMessage::GenerateCheckinMessagePayload(clientInfo.shared_key, counter, ByteSpan(),
+                                                                                        output);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    buffer->SetDataLength(static_cast<uint16_t>(output.size()));
+    ICDClientInfo decodeClientInfo;
+    uint32_t checkInCounter = 0;
+    ByteSpan payload{ buffer->Start(), buffer->DataLength() };
+    err = manager.ProcessCheckInPayload(payload, decodeClientInfo, checkInCounter);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    // 2. Use a key not available in the storage for encoding
+    err = manager.SetKey(clientInfo, ByteSpan(kKeyBuffer2));
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    err = chip::Protocols::SecureChannel::CheckinMessage::GenerateCheckinMessagePayload(clientInfo.shared_key, counter, ByteSpan(),
```

Comment:
I think this is the reason why OpenIoT sdk fails. You cannot use the same psa key ID for AES-CCM and HMAC operation.
For this to pass to you need #30926 and the associated changes.

Reply:
Created a ticket to track this